helix-loader/Cargo.toml

```diff
@@ -13,7 +13,7 @@ homepage = "https://helix-editor.com"
 anyhow = "1"
 serde = { version = "1.0", features = ["derive"] }
 toml = "0.5"
-etcetera = "0.3"
+directories = "4"
```

Comment:
❌ etcetera was specifically chosen because it uses XDG on macOS instead of system directories. For a command line tool it makes more  sense to use XDG.

Reply:
`directories`/`dirs` is also deprecated in favour of dirs-next (used by etcetera) https://www.reddit.com/r/rust/comments/pps4f0/directories_crate_version_400_broken_on_macos/

Reply:
What do you mean by system directories? Going by directories' README and etcetera code, all paths used on macOS begin with `$HOME/Library`. Both crates behave the same way.

`directories` crate isn't deprecated. It has more recent release than the *-next forks or etcetera and it's the only cross-platform crate I found that bothered to incorporate `XDG_STATE_HOME`. The Reddit link says nothing about deprecation either, just random people recommending directories-next, one without explanation and other saying "might be good alternative". It may have been at the time but now it's *-next that's outdated.

I'm open to using whatever, but etcetera and directories-next aren't up-to-date and don't do what's required.

Reply:
This is one of the defining features of etcetera: https://docs.rs/etcetera/latest/etcetera/base_strategy/fn.choose_base_strategy.html

Meanwhile it's not implemented in directories-next: https://github.com/xdg-rs/dirs/issues/45

directories had a minor release last year but it's unmaintained. Most crates switched over to directories-next after original author stopped responding: https://www.reddit.com/r/rust/comments/gli4pc/directoriesnext_dirsnext_the_new_home_for/

Reply:
Got it, I just looked at the Apple strategy. I created a PR https://github.com/arzg/etcetera/pull/1 to support state. Etcetera dev seems fairly active so hopefully it'll be merged soon.